File: work_optimization/src/main/java/wo/work_optimization/core/domain/entity/TaskEntity.java
package wo.work_optimization.core.domain.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "task")
public class TaskEntity {

    @Id
    @GeneratedValue(generator = "uuid")
    private String id;
    private String title;
    private int priority;
    private String status;
    private long startDate;
    private double duration;
    private long endDate;
    private boolean activeStatus;
    private String originalId;
}

File: person_task_manager/server/src/core/domain/entities/task.entity.ts
import { ICommentEntity } from "./comment.entity";
import { IUserTagEntity } from "./user-tag.entity";
import { ISubTaskEntity } from "./sub-task.entity";
import { ActiveStatus } from "../../../core/domain/enums/enums";

export interface ITaskEntity extends Document {
    _id: string;
    title: string;
    description: string;
    priority: string[];
    status: string;
    startDate: Date;
    deadline: Date;
    duration: number;
    subTasks: ISubTaskEntity["_id"][];
    comments: ICommentEntity["_id"][];
    createdAt: Date;
    updatedAt: Date;
    activeStatus: ActiveStatus;
    tag: IUserTagEntity 
}

